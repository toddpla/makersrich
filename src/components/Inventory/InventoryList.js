import React, { Component } from 'react';
import InventoryListItem from './InventoryListItem'

class InventoryList extends Component {

  render() {

    return (
      <div>
        {this.props.inventory.ruby.map((item, i) => <InventoryListItem key={i} item={item} />)}
        {this.props.inventory.bean.map((item, i) => <InventoryListItem key={i} item={item} />)}
        {this.props.inventory.key.map((item, i) => <InventoryListItem key={i} item={item} />)}
      </div>
    );
  }

}

export default InventoryList;

// let newArr = [];
//
// while(allItems.length) newArr.push(allItems.splice(0,2));

// <table>
//   <tbody>
//     {newArr.map((row, i) => (
//       <tr>
//         {row.map((cell, k) => (
//           <td>
//             {cell}
//           </td>
//         ))}
//       </tr>
//     ))}
//   </tbody>
// </table>
