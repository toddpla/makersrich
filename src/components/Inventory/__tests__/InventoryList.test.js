import React from 'react';
import { shallow } from 'enzyme'
import InventoryList from '../InventoryList'
import player from '../../../test/fixtures/playerWithInventory'

let wrapper

beforeEach(function() {
  const inventory = player.inventory
  wrapper = shallow(
    <InventoryList
      inventory={inventory}
    />
  )
});

test('it renders InventoryList', () => {
  expect(wrapper).toMatchSnapshot();
})
