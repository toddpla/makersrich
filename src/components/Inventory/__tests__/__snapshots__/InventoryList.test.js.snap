// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders InventoryList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InventoryList
    inventory={
      Object {
        "bean": Array [
          Object {
            "properties": Array [
              Object {
                "value": "This is a bean",
              },
            ],
            "type": "bean",
          },
        ],
        "key": Array [
          Object {
            "type": "key",
          },
        ],
        "miscellaneous": Array [
          Object {
            "properties": Array [
              Object {
                "value": "This is a ruby",
              },
            ],
            "type": "ruby",
          },
        ],
        "ruby": Array [
          Object {
            "properties": Array [
              Object {
                "value": "This is a ruby",
              },
            ],
            "type": "ruby",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Inventory
        </h1>,
        Array [
          <InventoryListItem
            item={
              Object {
                "properties": Array [
                  Object {
                    "value": "This is a ruby",
                  },
                ],
                "type": "ruby",
              }
            }
          />,
        ],
        Array [
          <InventoryListItem
            item={
              Object {
                "properties": Array [
                  Object {
                    "value": "This is a bean",
                  },
                ],
                "type": "bean",
              }
            }
          />,
        ],
        Array [
          <InventoryListItem
            item={
              Object {
                "type": "key",
              }
            }
          />,
        ],
        Array [
          <InventoryListMiscItem
            item={
              Object {
                "properties": Array [
                  Object {
                    "value": "This is a ruby",
                  },
                ],
                "type": "ruby",
              }
            }
          />,
        ],
      ],
      "id": "inventory-list-box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Inventory",
        },
        "ref": null,
        "rendered": "Inventory",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "handleMessage": undefined,
          "item": Object {
            "properties": Array [
              Object {
                "value": "This is a ruby",
              },
            ],
            "type": "ruby",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "handleMessage": undefined,
          "item": Object {
            "properties": Array [
              Object {
                "value": "This is a bean",
              },
            ],
            "type": "bean",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "handleMessage": undefined,
          "item": Object {
            "type": "key",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "handleMessage": undefined,
          "item": Object {
            "properties": Array [
              Object {
                "value": "This is a ruby",
              },
            ],
            "type": "ruby",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Inventory
          </h1>,
          Array [
            <InventoryListItem
              item={
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a ruby",
                    },
                  ],
                  "type": "ruby",
                }
              }
            />,
          ],
          Array [
            <InventoryListItem
              item={
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a bean",
                    },
                  ],
                  "type": "bean",
                }
              }
            />,
          ],
          Array [
            <InventoryListItem
              item={
                Object {
                  "type": "key",
                }
              }
            />,
          ],
          Array [
            <InventoryListMiscItem
              item={
                Object {
                  "properties": Array [
                    Object {
                      "value": "This is a ruby",
                    },
                  ],
                  "type": "ruby",
                }
              }
            />,
          ],
        ],
        "id": "inventory-list-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Inventory",
          },
          "ref": null,
          "rendered": "Inventory",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "handleMessage": undefined,
            "item": Object {
              "properties": Array [
                Object {
                  "value": "This is a ruby",
                },
              ],
              "type": "ruby",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "handleMessage": undefined,
            "item": Object {
              "properties": Array [
                Object {
                  "value": "This is a bean",
                },
              ],
              "type": "bean",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "handleMessage": undefined,
            "item": Object {
              "type": "key",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "handleMessage": undefined,
            "item": Object {
              "properties": Array [
                Object {
                  "value": "This is a ruby",
                },
              ],
              "type": "ruby",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
